Fix Mac JSCOnly build

emw-apple opened the pull request:
#### 74971bf54ad621a364b4c9af855ddba1176ba6e0
<pre>
Fix Mac JSCOnly build
<a href="https://bugs.webkit.org/show_bug.cgi?id=257148">https://bugs.webkit.org/show_bug.cgi?id=257148</a>
rdar://109674456

Reviewed by Alexey Proskuryakov.

OSLogPrintStream is now an objc++ file. Reflect this in CMakeLists.

As the JSCOnly port does not have a PLATFORM() macro, platform-derived
flags like USE(CF) are not available even though the port now does objc
compilation on macOS. To compensate, wrap use of Foundation types
(namely NSString) in USE(FOUNDATION). Both macros are based on
PLATFORM(COCOA).

* Source/WTF/wtf/CMakeLists.txt:
* Source/WTF/wtf/text/AtomString.h:
* Source/WTF/wtf/text/WTFString.h:

Canonical link: <a href="https://commits.webkit.org/264607@main">https://commits.webkit.org/264607@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/036f8ba94bdb84e59cc41807e041c1969a911f4f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/7648 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/7917 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/8101 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/9291 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/7828 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/9916 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/7842 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/10697 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/7781 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/8921 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/7046 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/9401 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/6229 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/6968 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/14649 "4 flakes 111 failures") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/6528 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/7392 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/7091 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/10500 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/7239 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/7584 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/6187 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/7803 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/6918 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/1786 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1949 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/11129 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/8010 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/7315 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/1930 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

ddkilzer commented on Source/WTF/wtf/text/AtomString.h:
```diff
@@ -111,12 +111,12 @@ class AtomString final {
     AtomString(NSString *);
     operator NSString *() const { return m_string; }
 #endif
+#endif
```
This doesn't look right.  If you want this code to be available only on Apple platforms, you should probably use `OS(DARWIN)` around both `#if USE(CF)` and `#ifdef __OBJC__`.

emw-apple committed:
Fix Mac JSCOnly build
https://bugs.webkit.org/show_bug.cgi?id=257148
rdar://109674456

Reviewed by Alexey Proskuryakov.

OSLogPrintStream is now an objc++ file. Reflect this in CMakeLists.

As the JSCOnly port does not have a PLATFORM() macro, platform-derived
flags like USE(CF) are not available even though the port now does objc
compilation on macOS. To compensate, wrap use of Foundation types
(namely NSString) in USE(FOUNDATION). Both macros are based on
PLATFORM(COCOA).

* Source/WTF/wtf/CMakeLists.txt:
* Source/WTF/wtf/text/AtomString.h:
* Source/WTF/wtf/text/WTFString.h:

Canonical link: https://commits.webkit.org/264607@main
